### PR TITLE
Fix/update several existing tests

### PR DESCRIPTION
**Pull request by dagardner-nv**
## Description
* Fix handling of union types in CLI
* Fix handling of file paths in RSS source
* Update tests to match recent code changes

## By Submitting this PR I confirm:
- I am familiar with the [Contributing Guidelines](https://github.com/nv-morpheus/Morpheus/blob/main/docs/source/developer_guide/contributing.md).
- When the PR is ready for review, new or existing tests cover these changes.
- When the PR is ready for review, the documentation is up to date with these changes.

### CODE REVIEW

**Review by mdemoret-nv — APPROVED** [review]

File: morpheus/cli/register_stage.py [hunk 1]
```diff
@@ -177,6 +178,10 @@ def set_options_param_type(options_kwargs: dict, annotation, doc_type: str):
     if (annotation == inspect.Parameter.empty):
         raise RuntimeError("All types must be specified to auto register stage.")
 
+    if (is_union_type(annotation)):
+        # For the purposes of the CLI, we treat unions as the first type
+        annotation = typing.get_args(annotation)[0]
+
```
[the remark this answers is not in the document]
Comment (dagardner-nv) [reply morpheus/cli/register_stage.py:184]: Currently in `fea-sherlock` running:
```
morpheus run pipeline-other --help
```

Will fail with an error:
```
RuntimeError: Error auto registering CLI command 'deserialize' with class '<class 'morpheus.stages.preprocess.deserialize_stage.DeserializeStage'>' and parameter 'message_type'. Error: issubclass() arg 1 must be a class
```

Telling the CLI to ignore the 'message_type' will still get us an error on the union type for the rss source stage.

File: morpheus/stages/input/rss_source_stage.py [hunk 2]
```diff
@@ -37,7 +37,7 @@ class RSSSourceStage(PreallocatorMixin, SingleOutputSource):
     ----------
     c : morpheus.config.Config
         Pipeline configuration instance.
-    feed_input : str
+    feed_input : list[str] or str
```
[the remark this answers is not in the document]
Comment (dagardner-nv) [reply morpheus/stages/input/rss_source_stage.py:40]: I believe `list[str] | str` works, I was following the numpydoc examples for this which uses an `or`:
https://numpydoc.readthedocs.io/en/latest/format.html#parameters

File: morpheus/cli/register_stage.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mdemoret-nv) [reply morpheus/cli/register_stage.py:184]: Lets switch the constructor to use only `list[str]`. And then add a check here to test for union types and show an error message that they are not supported for cli arguments

File: morpheus/stages/input/rss_source_stage.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mdemoret-nv) [reply morpheus/stages/input/rss_source_stage.py:40]: Or work then.

**Pull request merged**